fix README.md

hyuraku opened the pull request:
README.mdに各テーブルの情報と関係性について記述した。

以下テーブル名を記載する。
- users
- group_users
- groups
- messages
- comments

hyuraku committed:
fix README.md

README.mdに各テーブルの情報と関係性について記述した。
以下テーブル名を記載する。
- users
- group_users
- groups
- messages
- comments

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,76 @@
+Column | Type   | Options
+:----- | :----- | :----------------------------------
+name   | string | index:true, null: false, unique:true
+email  | text   | null: false
```
emailは文字数制限のあるstring型にしましょう。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,76 @@
+- has_many :groups ,though: group_users
+- has_many :messages
+- has_many :group_users
+- has_many :comments
```
comments必要ですか？

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,76 @@
+## groups table
+Column   | Type    | Options
+:------- | :------ | :-----------------------------
+name | string | index:true, null: false, unique:true
```
indexオプションがなぜ必要か、検討していただけると助かります。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,76 @@
+Column   | Type    | Options
+:------- | :------ | :-----------------------------
+name | string | index:true, null: false, unique:true
+user_id  | integer | foreign_key: true
```
groupsのテーブルにuser_id必要か、検討していただけると助かります。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,76 @@
+
+Column   | Type    | Options
+:------- | :------ | :-----------------------------
+body     | text    | null: false
```
chatという性質上、文字数制限のあるstring型にしましょう。また文字もしくは画像があれば投稿できる仕様ですのでnull: falseは外しましょう。

MrUrakou reviewed (commented):
commentsテーブルが、どのことを指しているのか、お手本のアプリを見ながら検証していただけると良いと思います。

hyuraku committed:
fix README.md

以下の4点を修正した。

・userテーブルのemailの型をtextから stringに変更
・commentsテーブルの削除、それに伴うアソシエーションの削除
（messageに対するcommentが仕様上必要なかったため）
・groupテーブルのnameのoptionよりindex:trueを削除
（該当テーブルに格納されるデータの種類がすくないため、indexは不要を考えた）
・groupsテーブルより、user_idを削除
(groupとuserの関係はgroup_userのテーブルで管理しているので、groupテーブル内にuserの情報は不要と考えた）
・messageテーブルのbodyの型をtextからstringに変更、null:falseの設定を削除

hyuraku committed:
Revert "fix README.md"

This reverts commit 74198a8ea2d22b1c96af27da67c5bf08e610e5d5.

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,60 @@
-* Database initialization
+Column   | Type    | Options
+:------- | :------ | :-----------------------------
+user_id  | integer | null: false, foreign_key: true
```
user_idもindexオプション付けましょう。最後にuserの検索機能を付けるので。

hyuraku committed:
fix README.md

add index: true to user_id option in group_users table
